plugins/builds/index.js

```diff
@@ -873,16 +1020,44 @@ const buildsPlugin = {
                         return createInternalBuild(internalBuildConfig);
                     }
 
-                    if (!['CREATED', null, undefined].includes(existNextBuild.status)) {
+                    if (!['CREATED', null, undefined].includes(existNextBuild.status) && !current.stage) {
                         return existNextBuild;
                     }
 
+                    // If currentBuild is stage teardown with success, check stageBuild status
+                    if (current.stage) {
+                        const stageBuild = await stageBuildFactory.get({
+                            stageId: current.stage.id,
+                            eventId: current.event.id
+                        });
+                        const stageTeardownName = getFullStageJobName({
+                            stageName: current.stage.name,
+                            jobName: 'teardown'
+                        });
+
+                        // If stageBuild status is not CREATED or does not exist, remove next build
+                        if (current.job.name === stageTeardownName && current.build.status === 'SUCCESS') {
+                            if (!stageBuild.status || stageBuild.status !== 'CREATED') {
+                                logger.info(
+                                    `Failure occurred in upstream job, removing new build - build:${existNextBuild.id} pipeline:${current.pipeline.id}-${current.job.name} event:${existNextBuild.eventId} `
+                                );
+                                await existNextBuild.remove();
+
+                                return null;
+                            }
+
+                            return existNextBuild;
+                        }
+                    }
+
+                    // Current build is not part of stage
```

Comment:
As suggested here https://github.com/screwdriver-cd/screwdriver/pull/2870/files/9f679937feaa4e278d80b3b1ef245984c9b03993#r1448111687, this block of code is no longer needed.

```suggestion
```